warp10/src/main/java/io/warp10/script/WarpScriptStack.java

```diff
@@ -739,14 +744,14 @@ private void ensureCapacity(int n) {
   public String getUUID();
   
   /**
-   * Abort the stack, i.e. stop the currently executing code after the current statement and prevent further executions.
+   * Signal the stack, i.e. stop the currently executing code after the current statement and prevent further executions.
    */
-  public void abort();
+  public void signal(Signal signal);
   
   /**
-   * Return the state of execution abortion.
+   * Throw the exception associated with the current signal sent to the stack
    */
-  public boolean aborted();
+  public void signal() throws WarpScriptATCException;
```

Comment:
I find `signal()` to be confusing, can it be called `checkSignal()`?

Reply:
It's not a check, it's a triggering of a buffered signal, so maybe triggerSignal?